File: article/src/main/java/com/ruzhan/awaker/article/imageloader/ImageLoader.java
package com.ruzhan.awaker.article.imageloader;

import com.ruzhan.awaker.article.imageloader.glide.GlideImpl;

public final class ImageLoader {

    private static IImageLoader imageLoader;

    private ImageLoader() {
    }

    public static IImageLoader get() {
        if (imageLoader == null) {
            synchronized (ImageLoader.class) {
                if (imageLoader == null) {
                    imageLoader = new GlideImpl();
                }
            }
        }
        return imageLoader;
    }
}

File: article/src/main/java/com/ruzhan/awaker/article/imageloader/ImageLoader.kt
package com.ruzhan.awaker.article.imageloader

import com.ruzhan.awaker.article.imageloader.glide.GlideImpl

object ImageLoader {

    private var imageLoader: IImageLoader? = null

    fun get(): IImageLoader {
        if (imageLoader == null) {
            synchronized(ImageLoader::class.java) {
                if (imageLoader == null) {
                    imageLoader = GlideImpl()
                }
            }
        }
        return imageLoader!!
    }
}
